test/python/pulse/test_commands.py

```diff
@@ -244,5 +245,103 @@ def test_can_construct_discriminator_with_default_values(self):
+        np.testing.assert_almost_equal(samples, gauss)
+
+    def test_gauss_square_samples(self):
+        """Test that the gaussian square samples match the formula."""
+        duration = 125
+        sigma = 4
+        amp = 0.5j
+        # formulaic
+        times = np.array(range(25), dtype=np.complex_)
+        times = times - (25 / 2) + 0.5
+        gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        # command
+        command = GaussianSquare(duration=duration, sigma=sigma, amp=amp, width=100)
+        samples = command.get_sample_pulse().samples
+        np.testing.assert_almost_equal(samples[50], amp)
+        np.testing.assert_almost_equal(samples[100], amp)
+        np.testing.assert_almost_equal(samples[:10], gauss[:10])
+
+    def test_drag_samples(self):
+        """Test that the drag samples match the formula."""
+        duration = 25
+        sigma = 4
+        amp = 0.5j
+        beta = 1
+        # formulaic
+        times = np.array(range(25), dtype=np.complex_)
+        times = times - (25 / 2) + 0.5
+        gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        gauss_deriv = -(times / sigma**2) * gauss
+        drag = gauss + 1j * beta * gauss_deriv
+        # command
+        command = Drag(duration=duration, sigma=sigma, amp=amp, beta=beta)
+        samples = command.get_sample_pulse().samples
+        np.testing.assert_almost_equal(samples, drag)
+
+    def test_constant_samples(self):
+        """Test the constant pulse and its sampled construction."""
+        const = ConstantPulse(duration=150, amp=0.1 + 0.4j)
+        self.assertEqual(const.get_sample_pulse().samples[0], 0.1 + 0.4j)
+        self.assertEqual(len(const.get_sample_pulse().samples), 150)
+
+    def test_get_params(self):
+        """Test that the parameters can be extracted as a dict through `get_params`."""
+        drag = Drag(duration=25, amp=0.2 + 0.3j, sigma=7.8, beta=4)
+        self.assertEqual(set(drag.get_params().keys()), {'duration', 'amp', 'sigma', 'beta'})
+        const = ConstantPulse(duration=150, amp=1)
+        self.assertEqual(set(const.get_params().keys()), {'duration', 'amp'})
+
+    def test_complex_param_is_complex(self):
+        """Check that complex param 'amp' is cast to complex."""
+        const = ConstantPulse(duration=150, amp=1)
+        self.assertIsInstance(const.amp, complex)
+
+    def test_param_validation(self):
+        """Test that parametric pulse parameters are validated when initialized."""
+        with self.assertRaises(PulseError):
+            Gaussian(duration=25, sigma=0, amp=0.5j)
+        with self.assertRaises(PulseError):
+            GaussianSquare(duration=150, amp=0.2, sigma=8, width=160)
+        with self.assertRaises(PulseError):
+            ConstantPulse(duration=150, amp=0.9 + 0.8j)
+        with self.assertRaises(PulseError):
+            Drag(duration=25, amp=0.2 + 0.3j, sigma=-7.8, beta=4)
+        with self.assertRaises(PulseError):
```

Comment:
Maybe worth adding one or two more tests-for negative duration, check that amp=1 doesn't throw error, etc. Up to you, but extra cases always good.